src/lib/storage/dictionary_column.hpp

```diff
@@ -24,48 +32,114 @@ class DictionaryColumn : public BaseColumn {
   /**
    * Creates a Dictionary column from a given value column.
    */
-  explicit DictionaryColumn(const std::shared_ptr<BaseColumn>& base_column);
-
-  // SEMINAR INFORMATION: Since most of these methods depend on the template parameter, you will have to implement
-  // the DictionaryColumn in this file. Replace the method signatures with actual implementations.
+  explicit DictionaryColumn(const std::shared_ptr<BaseColumn>& base_column) {
+    const auto value_column = dynamic_cast<ValueColumn<T>*>(base_column.get());
```

Comment:
You could use a `std::dynamic_pointer_cast`, so that you don't have to deal with raw pointers. However, I guess it should be safe either way because the base_column ref keeps another reference to the memory alive.